cmd/img_mgmt/src/img_mgmt.c

```diff
@@ -165,7 +165,15 @@ img_mgmt_read_info(int image_slot, struct image_version *ver, uint8_t *hash,
      * the hash is missing, the image is considered invalid.
      */
     data_off = hdr.ih_hdr_size + hdr.ih_img_size;
-    rc = img_mgmt_find_tlvs(&hdr, image_slot, &data_off, &data_end);
+
+    rc = img_mgmt_find_tlvs(image_slot, &data_off, &data_end, IMAGE_TLV_PROT_INFO_MAGIC);
+    if (rc != 0) {
+        return MGMT_ERR_EUNKNOWN;
+    }
+
+    data_off = data_end - sizeof(struct image_tlv_info);
```

Comment:
Personally I would add a brief comment here explaining that the semantics are different for protected TLV info vs unprotected.  It might look like a bug since the code only does this subtraction in one of two places.

Reply:
done